### PR TITLE
Rewrite overview to functional component & refactor

### PR DESCRIPTION
**Pull request by LudvigHz**
Rewrite overview component to be functional, refactor a lot of stuff and fix a bunch of types.

Also :star_struck: :star_struck::

[Screencast from 2023-01-18 13-10-12.webm](https://user-images.githubusercontent.com/42850232/213168103-0b5b0373-9c87-4f11-9bbb-9f8b35c2e9eb.webm)

### CODE REVIEW

File: app/routes/overview/components/Overview.tsx [hunk 1]
```diff
@@ -16,217 +16,227 @@ import styles from './Overview.css';
...
+    [articles, articlesToShow, pinned]
+  );
+
+  const pinnedComponent = pinned && (
+    <div>
+      <Pinned item={pinned} url={itemUrl(pinned)} meta={renderMeta(pinned)} />
+    </div>
+  );
+
+  const readMe = (
```
Comment (ivarnakken) [inline app/routes/overview/components/Overview.tsx:84]: 🤮

File: app/reducers/events.ts [hunk 2]
```diff
@@ -192,7 +195,7 @@ export const selectUpcomingEvents = createSelector(selectEvents, (events) =>
   events.filter((event) => event.isUsersUpcoming)
 );
 export const selectSortedEvents = createSelector(selectEvents, (events) =>
-  events.sort((a, b) => a.startTime - b.startTime)
+  [...events].sort((a, b) => a.startTime.unix() - b.startTime.unix())
```
Comment (ivarnakken) [inline app/reducers/events.ts:198]: What's the point of `[...events]`? Is `events` not already an array of `Event`?

File: app/components/RandomQuote/RandomQuote.css [hunk 3]
```diff
@@ -20,3 +20,8 @@
 .fetchNew {
   margin-bottom: 5px;
 }
+
+.rotateIcon {
+  transform: rotate(360deg);
+  transition: 1s;
```
Comment (ivarnakken) [inline app/components/RandomQuote/RandomQuote.css:26]: I feel like a `cubic-bezier` would be very fitting, in which it speeds up at the middle and slows down at the end. 

On that note, we should standardize a `cubic-bezier` for the webapp as a variable. I feel like animations across the webapp is very "random", with different animations, speeds and lengths. I'll write it up as a todo, so you can keep it like this if you want.

**Review by ivarnakken — APPROVED** [review]
Looks good! ⭐ A "better" overview component is going to be very handy!

I like the tiny, yet impactful, animation touch! 😍

File: app/reducers/events.ts (hunk 2 above)
Comment (LudvigHz) [reply app/reducers/events.ts:198]: JS `.sort` is in-place. Both being annoying as mutating arrays when doing things like this is a bad idea IMO, and the fact that `events` already has a type, and the result of the `.sort()` is something else, and TS can't magically change the type of a variable. So creating a copy ensures that TS infers the correct type.

File: app/components/RandomQuote/RandomQuote.css (hunk 3 above)
Comment (LudvigHz) [reply app/components/RandomQuote/RandomQuote.css:26]: I thought so too, but the animation was just sooo satisfying by default so I saw no reason to create more code for little to no benefit.

File: app/reducers/events.ts (hunk 2 above)
Comment (ivarnakken) [reply app/reducers/events.ts:198]: ah yes that makes perfect sense, I forgot that it was in-place.

File: app/components/RandomQuote/RandomQuote.css (hunk 3 above)
Comment (ivarnakken) [reply app/components/RandomQuote/RandomQuote.css:26]: Yea, I'll agree that it looks smooth

**Pull request merged**